packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -175,101 +158,28 @@ library LibVerifying {
-        TaikoData.ForkChoice memory fc,
-        uint24 fcId
+        TaikoData.ForkChoice memory fc
     )
         private
     {
-        // the actually mined L2 block's gasLimit is blk.gasLimit +
-        // LibL2Consts.ANCHOR_GAS_COST, so fc.gasUsed may greater than
-        // blk.gasLimit here.
-        uint32 _gasLimit = blk.gasLimit + LibL2Consts.ANCHOR_GAS_COST;
-        assert(fc.gasUsed <= _gasLimit);
-
-        IProverPool proverPool =
-            IProverPool(resolver.resolve("prover_pool", false));
-
-        if (blk.assignedProver == address(0)) {
-            --state.slotB.numOpenBlocks;
-        } else if (!blk.proverReleased) {
-            proverPool.releaseProver(blk.assignedProver);
-        }
-
-        // Reward the prover (including the oracle prover)
-        uint64 proofReward =
-            (config.blockFeeBaseGas + fc.gasUsed) * blk.rewardPerGas;
-
-        if (fc.prover == address(1)) {
-            // system prover is rewarded with `proofReward`.
-        } else if (blk.assignedProver == address(0)) {
-            // open prover is rewarded with more tokens
-            proofReward = proofReward * config.rewardOpenMultipler / 100;
-        } else if (blk.assignedProver != fc.prover) {
-            // proving out side of the proof window, by a prover other
-            // than the assigned prover
-            proofReward = proofReward * config.rewardOpenMultipler / 100;
-            proverPool.slashProver(blk.blockId, blk.assignedProver, proofReward);
-        } else if (fc.provenAt <= blk.proposedAt + blk.proofWindow) {
-            // proving inside the window, by the assigned prover
-            uint64 proofDelay;
-            unchecked {
-                proofDelay = fc.provenAt - blk.proposedAt;
-
-                if (config.rewardMaxDelayPenalty > 0) {
-                    // Give the reward a penalty up to a small percentage.
-                    // This will encourage prover to submit proof ASAP.
-                    proofReward -= proofReward * proofDelay
-                        * config.rewardMaxDelayPenalty / 10_000 / blk.proofWindow;
-                }
-            }
-
-            // The selected prover managed to prove the block in time
-            state.slotC.avgProofDelay = uint16(
-                LibUtils.movingAverage({
-                    maValue: state.slotC.avgProofDelay,
-                    newValue: proofDelay,
-                    maf: 7200
-                })
-            );
-
-            state.slotC.feePerGas = uint32(
-                LibUtils.movingAverage({
-                    maValue: state.slotC.feePerGas,
-                    newValue: blk.rewardPerGas,
-                    maf: 7200
-                })
-            );
+        if (
+            fc.prover == address(1)
+                || fc.provenAt <= blk.proposedAt + config.proofWindow
+        ) {
+            // Refund all the bond
+            tt.transfer(blk.prover, config.proofBond);
```

Comment:
If we'll keep a bond per block, than `proofBond` may have to be saved in the block because otherwise if `config.proofBond` increases between proposal and here, this may fail because too few tokens that can be transferred (because fewer were deposited at proposal).

Reply:
It's indeed an issue. How about we simply use burn/mint instead to avoid using one extra storage slot per block.

Reply:
Burn/mint is indeed easier, not sure what the cost difference is between transfer/burn/mint because burn/mint also need to update the total balance.